contracts/Registry.sol

```diff
@@ -99,12 +99,35 @@ contract Registry is AccessControlDefaultAdminRules {
     );
   }
 
-  function registerReferral(
-    string calldata referrerId,
-    string calldata protocolId
-  ) external {
-    _referrerIsRegistered = false;
-    for (uint256 i = 0; i < _referrerIdToProtocolIds[referrerId].length; i++) {
+  function getUserRegistrations(
+                                 address userAddress,
+                                 bytes32[] calldata protocolIds
+  ) external view returns (bool[] memory) {
+      bool[] memory registrations = new bool[](protocolIds.length);
+      for (uint256 i = 0; i < protocolIds.length; i++) {
+          registrations[i] = this.getUserRegistration(userAddress, protocolIds[i]);
+      }
+      return registrations;
+  }
+
+  function getUserRegistration(
+                                 address userAddress,
+                                 bytes32 protocolId
+  ) external view returns (bool) {
+      return _userInfoByProtocol[protocolId][userAddress].timestamp != 0;
+  }
```

Comment:
Given the arguments and return value, what do you think of this?
Also I think 1 function that takes an array is enough.

```suggestion
  function isUserRegistered(
                                 address userAddress,
                                 bytes32[] calldata protocolIds
  ) external view returns (bool[] memory) {
      bool[] memory registeredStatuses = new bool[](protocolIds.length);
      for (uint256 i = 0; i < protocolIds.length; i++) {
          registeredStatuses[i] = _userInfoByProtocol[protocolId][userAddress].timestamp != 0;
      }
      return registeredStatuses;
  }
```

In addition, this is slightly more efficient because there's no call (though doesn't matter that much for a view call).